plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteMetadata.java

```diff
@@ -123,6 +130,25 @@ public Optional<ConnectorOutputMetadata> finishInsert(
         return Optional.empty();
     }
 
+    @Override
+    public Optional<ConnectorTableHandle> applyUpdate(ConnectorSession session, ConnectorTableHandle handle, Map<ColumnHandle, Constant> assignments)
+    {
+        // Ignite support row level update, so we should reject this path, earlier than in JDBC client
+        return Optional.empty();
+    }
+
+    @Override
+    public void finishMerge(ConnectorSession session, ConnectorMergeTableHandle mergeTableHandle, List<ConnectorTableHandle> sourceTableHandles, Collection<Slice> fragments, Collection<ComputedStatistics> computedStatistics)
```

Comment:
Why do we have customize here?

Reply:
Ignite not support transaction write, no finish operation required.